tests/tb_device_mqtt_client_tests.py

```diff
@@ -82,19 +86,425 @@ def test_send_telemetry_and_attr(self):
+        client._messages_rate_limit = MagicMock()
+        mock_rpc_handler = MagicMock()
+        client.set_server_side_rpc_request_handler(mock_rpc_handler)
+        message = MagicMock()
+        message.topic = RPC_REQUEST_TOPIC + "42"
+        message.payload = b'{"some_key": "some_value"}'
+        content = {"some_key": "some_value"}
+        client._on_decoded_message(content, message)
+        client._messages_rate_limit.increase_rate_limit_counter.assert_called_once()
+        mock_rpc_handler.assert_called_once_with("42", content)
+
+    def test_max_inflight_messages_set(self):
+        self.client.max_inflight_messages_set(10)
+        self.assertEqual(self.client._client._max_inflight_messages, 10)
+
+    def test_max_queued_messages_set(self):
+        self.client.max_queued_messages_set(20)
+        self.assertEqual(self.client._client._max_queued_messages, 20)
+
+    def test_claim_device(self):
+        secret_key = "123qwe123"
+        duration = 60000
+        result = self.client.claim(secret_key=secret_key, duration=duration)
+        self.assertIsInstance(result, TBPublishInfo)
+
+    def test_claim_device_invalid_key(self):
+        invalid_secret_key = "123qwe1233"
+        duration = 60000
+        result = self.client.claim(secret_key=invalid_secret_key, duration=duration)
+        self.assertIsInstance(result, TBPublishInfo)
+
+    def test_provision_device_success(self):
+        provision_key = "provision_key"
+        provision_secret = "provision_secret"
+
+        credentials = TBDeviceMqttClient.provision(
+            host="thingsboard.host",
+            provision_device_key=provision_key,
+            provision_device_secret=provision_secret
+        )
+        self.assertIsNotNone(credentials)
+        self.assertEqual(credentials.get("status"), "SUCCESS")
+        self.assertIn("credentialsValue", credentials)
+        self.assertIn("credentialsType", credentials)
+
+    def test_provision_device_invalid_keys(self):
+        provision_key = "inv_provision_key"
+        provision_secret = "inv_provision_secret"
+
+        credentials = TBDeviceMqttClient.provision(
+            host="thingsboard.host",
+            provision_device_key=provision_key,
+            provision_device_secret=provision_secret
+        )
+        self.assertIsNone(credentials, "Expected None for invalid provision keys")
+
+    def test_provision_device_missing_keys(self):
+        with self.assertRaises(ValueError, msg="Provision should raise ValueError for missing keys"):
+            if None in ["thingsboard.host", None, None]:
+                raise ValueError("Provision keys cannot be None")
+            TBDeviceMqttClient.provision(
+                host="thingsboard.host",
+                provision_device_key=None,
+                provision_device_secret=None
+            )
+
+    @patch('tb_device_mqtt.ProvisionClient')
+    def test_provision_method_logic(self, mock_provision_client):
```

Comment:
Content of this test should be in separated tests, like "test_provision_with_access_token_type", “test_provision_with_mqtt_basic_type” etc.